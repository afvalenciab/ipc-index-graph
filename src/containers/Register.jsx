import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import '../assets/styles/Register.scss';

const Register = (props) => {

  const [form, setForm] = useState({
    email: '',
  });

  const handleInput = (event) => {
    setForm({
      ...form,
      [event.target.name]: event.target.value,
    });
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    console.log(form);
  };

  return (
    <main className='container__register'>
      <section className='register'>
        <h2>Crea tu registro</h2>
        <form onSubmit={handleSubmit}>
          <input name='email' type='email' placeholder='Correo' onChange={handleInput} required />
          <input name='password' type='Password' placeholder='Constraseña' onChange={handleInput} required />
          <button type='submit'>Registrarme</button>
        </form>
        <div className='register__social'>
          <a href='/register'>
            Registrarme con Google
          </a>
        </div>
        <div className='register__login'>
          <p>
            ¿Ya estás registrado?
            <Link to='/login'>
              Inicia sesión
            </Link>
          </p>
        </div>
      </section>
    </main>
  );
};

export default Register;
