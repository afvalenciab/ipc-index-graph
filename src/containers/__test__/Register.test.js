import React from 'react';
import { create } from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ProviderMock from '../../__mocks__/ProviderMocks';
import Register from '../Register';

configure({ adapter: new Adapter() });

describe('Register testing', () => {
  test('Match Snapshot', () => {
    const register = create(
      <ProviderMock>
        <Register />
      </ProviderMock>
    );
    expect(register.toJSON()).toMatchSnapshot();
  });

  test('Calls and execute preventDefault function onSubmit form', () => {
    const preventDefault = jest.fn();
    const register = mount(
      <ProviderMock>
        <Register />
      </ProviderMock>
    );
    register.find('form').simulate('submit', { preventDefault });
    expect(preventDefault).toHaveBeenCalledTimes(1);
    register.unmount();
  });
});
