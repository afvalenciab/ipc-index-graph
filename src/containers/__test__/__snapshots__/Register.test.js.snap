// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register testing Match Snapshot 1`] = `
<main
  className="container__register"
>
  <section
    className="register"
  >
    <h2>
      Crea tu registro
    </h2>
    <form
      onSubmit={[Function]}
    >
      <input
        name="email"
        onChange={[Function]}
        placeholder="Correo"
        required={true}
        type="email"
      />
      <input
        name="password"
        onChange={[Function]}
        placeholder="Constraseña"
        required={true}
        type="Password"
      />
      <button
        type="submit"
      >
        Registrarme
      </button>
    </form>
    <div
      className="register__login"
    >
      <p>
        ¿Ya estás registrado?
        <a
          href="/login"
          onClick={[Function]}
        >
          Inicia sesión
        </a>
      </p>
    </div>
  </section>
</main>
`;
